tests/test_stnode.py

```diff
@@ -113,6 +113,18 @@ def test_serialization(node_class, tmp_path):
         assert_node_equal(af["node"], node)
 
 
+@pytest.mark.parametrize("node_class", stnode.NODE_CLASSES)
+@pytest.mark.filterwarnings("ignore:Input shape must be 1D")
+@pytest.mark.filterwarnings("ignore:This function assumes shape is 2D")
+@pytest.mark.filterwarnings("ignore:Input shape must be 4D")
+@pytest.mark.filterwarnings("ignore:Input shape must be 5D")
+def test_no_hidden(node_class):
+    if issubclass(node_class, stnode.TaggedObjectNode):
```

Comment:
```suggestion
    if issubclass(node_class, (stnode.TaggedObjectNode | stnode.TaggedListNode)):
```
Would you move this filtering into the parameterization?

Reply:
This does not apply to list nodes.

Reply:
Would you remove the closing verbiage for https://github.com/spacetelescope/roman_datamodels/issues/418 and update it to reflect that the bug still applies for list nodes?

Reply:
#418 does not indicate any application to list nodes nor would one expect serialization of arbitrary attributes attached to a list to be serialize.

I will add this but I don't think it is necessary

Reply:
Thanks. That issue wasn't specific to any node type. It also doesn't have anything to do with serialization. Here's an example with this PR where a private attribute assignment is lost:
```
import roman_datamodels.datamodels as rdm
m = rdm.ImageModel.create_fake_data()
m.meta.exposure.read_pattern._foo = 1
assert m.meta.exposure.read_pattern._foo == 1
```
Fails with:
```
AttributeError: 'LNode' object has no attribute '_foo'
```
If we're going to prevent private attributes on `DNode` it makes sense to me to enforce the same behavior on `LNode`.